### PR TITLE
cameleon-device: remove emulator code

### PR DESCRIPTION
**Pull request by strohel**
This is commented-out from compilation and doesn't compile if uncommented.

Please revert this commit if there's an interest in continuing this effort.

- [x] Check `test_all.sh` is passed.
- [x] Add `fix #{ISSUE_NUMBER}` if the corresponding issue exists.
- [x] Fill out `## Changelog` section. If the change is for only internal use, please write `None` to the section.

## Changelog
None

### CODE REVIEW

File: device/README.md [hunk 1]
```diff
@@ -14,3 +14,7 @@
 
 ## Overview
 `cameleon-device` provides device specific protocol decoder and basic I/O operations for [GenICam](https://www.emva.org/standards-technology/genicam/) compatible devices, also provides emulators.
+
+### Cargo Features
+
+- `emulator`: work-in-progress emulator code, uncomment in lib.rs to resume working on it.
```
Comment (bschwind) [inline device/README.md:20]:
```suggestion
- `emulator`: work-in-progress emulator code, uncomment in `device/src/lib.rs` to resume working on it.
```

**Review by bschwind — APPROVED** [review]
LGTM with a small nitpick

**Comment by Y-Nak** [comment]
It's fine for me to remove the entire `emulator` code. I guess no one will use the feature.
If it turns out that it's useful, we can revive it from the commit.

**Comment by strohel** [comment]
@Y-Nak good point, I forgot we've got git. :wink:

I've turned this into a PR that removes the code, PTAL (please take another look). CC @bschwind.

**Comment by strohel** [comment]
Rebased to resolve conflict.

**Review by bschwind — APPROVED** [review]
Code looks good, though I feel bad about deleting a bunch of work (I know the code is alive in git).

@Y-Nak how far away was this code from working? I'm assuming it was written to run cameleon tests without needing a camera connected, is that correct?

**Comment by Y-Nak** [comment]
👍

**Pull request merged**